client/components/phone-input/index.tsx

```diff
@@ -310,68 +331,55 @@ function shouldGuessCountry( value, countryCode, countriesList, freezeSelection
  *
  * @param {string} value - Input number
  * @param {string} fallbackCountryCode - Fallback country code in case we can't find a match
- * @param {Array} countriesList - The list of countries
  * @param {boolean} freezeSelection - True if we should never guess the country
- * @returns {{isoCode: string, dialCode: string, nationalPrefix: string}} - Country metadata
  */
 function guessCountryFromValueOrGetSelected(
-	value,
-	fallbackCountryCode,
-	countriesList,
-	freezeSelection
+	value: string,
+	fallbackCountryCode: string,
+	freezeSelection: boolean
 ) {
-	if ( shouldGuessCountry( value, fallbackCountryCode, countriesList, freezeSelection ) ) {
-		return findCountryFromNumber( value ) || getCountry( 'world', countriesList );
+	if ( shouldGuessCountry( value, fallbackCountryCode, freezeSelection ) ) {
+		return findCountryFromNumber( value ) || getCountry( 'world' );
 	}
 
-	return getCountry( fallbackCountryCode, countriesList );
+	return getCountry( fallbackCountryCode );
 }
 
 /**
  * Format the phone number for display
  *
  * @param {string} value - The phone number value to format
  * @param {string} countryCode - The country code to use for the format
- * @param {Array} countriesList - The country data
  * @returns {string} The formatted number
  */
-function format( value, countryCode, countriesList ) {
-	return formatNumber( value, getCountry( countryCode, countriesList ) );
+function format( value: string, countryCode: string ): string {
+	return formatNumber( value, getCountry( countryCode ) );
 }
 
 /**
  * Returns the country meta with default values for countries with missing metadata. Never returns null.
- *
- * @param {string} countryCode - The country code
- * @param {Array} countriesList - The country data
- * @returns {{isoCode: string, dialCode: string, nationalPrefix: string}} - Country metadata
  */
-function getCountry( countryCode, countriesList ) {
-	let selectedCountry = countries[ countryCode ];
+function getCountry( countryCode: string ) {
+	let selectedCountry = countries[ countryCode as keyof typeof countries ];
 
 	if ( ! selectedCountry ) {
-		// Special cases where the country is in a disputed region and not globally recognized.
-		// At this point this should only be used for: Canary islands, Kosovo, Netherlands Antilles
-		const data = find( countriesList || [], ( { code } ) => code === countryCode );
-
 		selectedCountry = {
 			isoCode: countryCode,
-			dialCode: ( ( data && data.numeric_code ) || '' ).replace( '+', '' ),
```

Comment:
This was removed because as far as I can tell, there's no such property as `numeric_code` anywhere within country data. I searched both the front-end and the back-end for that property and the only place I found it was in the `/meta/sms-country-codes` API endpoint, which I don't believe is ever used with the phone number input.

As a result of removing this, a ton of the functions which required `countriesList` as an argument no longer require it.